cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/DiffLogic.java

```diff
@@ -0,0 +1,70 @@
+package cloud.foundry.cli.logic;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import cloud.foundry.cli.crosscutting.exceptions.DiffException;
+import cloud.foundry.cli.crosscutting.mapping.beans.Bean;
+import cloud.foundry.cli.logic.diff.DiffNode;
+import cloud.foundry.cli.logic.diff.Differ;
+import cloud.foundry.cli.logic.diff.output.DiffOutput;
+
+/**
+ * Handles the operations to compare the configuration of a cloud foundry instance with a different configuration.
+ */
+public class DiffLogic {
+
+    /**
+     * Compares the two given configurations and creates a tree composed of @DiffNode objects.
+     * @param liveConfig the configuration that is currently on the live system
+     * @param desiredConfig the configuration state that the live system should change to
+     * @return @DiffNode object which is the root of the tree
+     * @throws DiffException when liveConfig or desiredConfig is null,
+     * when the two beans don't have the same type
+     * or in case of any errors during the diff procedure
+     */
+    public DiffNode createDiffTree(Bean liveConfig, Bean desiredConfig) throws DiffException {
+        try {
+            checkNotNull(liveConfig);
```

Comment:
I'm pretty sure that a `NullPointerException` due to a failing precondition check should *not* be wrapped into a `DiffException`.